tests/draft-next/optional/format/time.json

```diff
@@ -41,6 +41,11 @@
                 "data": "08:30:06Z",
                 "valid": true
             },
+            {
+                "description": "hour, minute, second must be two digits",
+                "data": "8:0030:6Z",
```

Comment:
I think this looks good to me too behavior-wise from the spec, but while we're here I think it's a bit less fragile to split this test in 2, one to ensure an implementation doesn't allow extra digits (leading zeros) and one that it requires a leading zero for single digits.

This specific test may fail for one reason and not the other.

Reply:
I fabricated this test case to check my implementation in rust:

my implementation works as follows:
1. reject if string length is less than 9
2. split first 8 characters by `:`
3. ensure you get 3 tokens and parse each token as integer into `h` `m` and `s`
4. check `h` `m` `s` are in correct range

splitting the test into two cases as you said, does not work with the above implementation. since only both combinations will fail for my case.

Reply:
Fair enough, perhaps leave it in place and add 2 extra then?

Reply:
ok

Reply:
> splitting the test into two cases as you said, does not work with the above implementation. since only both combinations will fail for my case.

Wouldn't that imply that your implementation is incomplete, then?

Reply:
@karenetheridge test says`1:2:3z` should fail because there is no leading zero before single digit.
my implementation also says it is invalid, but the reason is its length less than `9`
so if some one constructs a time string length >=9 with no leading zero before single digit then my implementation will say it is valid( but actually it is invalid).

Reply:
> if some one constructs a time string length >=9 with no leading zero before single digit then my implementation will say it is valid( but actually it is invalid).

Let's add a test case for that then, as it's pretty tricky!

Reply:
@karenetheridge the tescase for that is already included in this PR